modules/schedule/index.js

```diff
@@ -0,0 +1,396 @@
+      reply("       Also see !schedule help <subcommand> for more details");
+    },
+    help: {
+      _default: null, // to be aliased later to ^
+      add: function(x,y,reply) {
+        reply("Usage: !schedule add \"timeframe\" \"command or text\"");
+        reply("       Timeframe syntax is here: http://bunkat.github.io/later/parsers.html#text");
+        reply("       If a valid command isn't specified, it is treated as text to print.");
+      },
+      remove: function(x,y,reply) {
+        reply("Usage: !schedule remove [digest]");
+        reply("       Removes the schedule specified by the digest.");
+      },
+      list: function(x,y,reply) {
+        reply("Usage: !schedule list [offset=0]");
+        reply("       Provides a list of schedules, in pages, from the specified offset of 0.");
+      },
+    },
+    add: function(r, parts, reply, command, from, to, text, raw) {
+      if(parts.length !== 2) return reply("add must have *exactly* two arguments");
+
+      reply(newSchedule({
+        'blame': from,
+        'created': new moment(),
+        'schedule': parts[0],
+        'command': parts[1],
+        'channel': to,
+        'calls': -1
+      }));
+    },
+    list: function(x, parts, reply, command, from, to) {
+      var offset = 0;
+      var count  = 5;
+      var ci     = count;
+      var oi     = 0;
+      var size = Object.keys(schedules).length;
+
+      if(parts.length >= 1)
+        oi = offset = parseInt(parts[0]);
+
+      if(offset < 0)
+        oi = offset = size + offset;
+
+      if(offset >= size)
+        return bot.sayTo(from, "There are only " + size + " schedules");
+      
+      if(size == 0)
+        return bot.sayTo(from, "There are no schedules.");
+
+      if(count+offset > size)
+        count = size-offset;
+
+      // count and pick
+      Object.keys(schedules).forEach(function(digest) {
+        if(oi > 0)
+        {
+          oi -= 1;
+          return;
+        }
+
+        if(ci == 0)
+          return;
+
+        var e = schedules[digest];
+        var channel = e.channel;
+
+        if(!channel.match(/^(#|&)/))
+          channel = "@"+channel;
+
+        bot.sayTo(from, digest + "     " + e.blame + "     " 
+                + e.created.format("ddd MM/DD/YY HH:mm:ss Z")
+                + "     " + channel);
+
+        var message =  "     ";
+        if( e.command.match(/^!/) )
+          message += "command: " + e.command;
+        else
+          message += "say: " + e.command;
+        bot.sayTo(from, message);
+
+        if(ci > 0)
+          ci -= 1;
+      });
+
+      var message = "Displayed schedules " + (offset+1) + "-";
+      message += (offset+count) + " of " + size;
+
+      bot.sayTo(from, message);
+    },
+
+    remove: function(r, parts, reply) {
+      if(parts.length !== 1) return reply("remove must have *exactly* one argument");
+      
+      var digest = parts[0];
+
+      if(schedules.length == 0)
```

Comment:
This is never true because `schedules` is an object, so `schedules.length` is `undefined`.

If you want to make schedules a real object with a length getter, feel free, I have no objection to fancy new js features.